### PR TITLE
Use multi search api for ms v1.1.0

### PR DESCRIPTION
**Pull request by bidoubiwa**
Replace the usage of `client.index().search` with `client.multiSearch()` when performing searches.

- TODO: 
- [x] Replace search requests made with `client.index.search` with `client.multiSearch`
   - Pagination
   - Caching
- [x] Use multi-search also when performing the initial facet distribution
- [x] Rename `searchResponse` with `searchResult` when it is not the whole response of the multi search API

### CODE REVIEW

File: packages/instant-meilisearch/src/client/config/index.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    )
+  }
+
+  // Validate api key
+  if (typeof apiKey !== 'string' && typeof apiKey !== 'function') {
+    throw new TypeError(
+      'Provided apiKey value (2nd parameter) is not a string or a function, expected string or function'
+    )
+  }
+}
```
Comment (bidoubiwa) [inline packages/instant-meilisearch/src/client/config/index.ts:71]: no need to review this. They have been moved from another file to this file.

File: packages/instant-meilisearch/src/client/instant-meilisearch-client.ts [hunk 2]
```diff
@@ -120,24 +132,3 @@ export function instantMeiliSearch(
...
-      throw new TypeError(
-        'Provided apiKey function (2nd parameter) did not return a string, expected string'
-      )
-    }
-
-    return apiKeyFnValue
-  }
-
-  return apiKey
-}
```
Comment (bidoubiwa) [inline packages/instant-meilisearch/src/client/instant-meilisearch-client.ts:143]: moved to `./config/index.ts`

File: .changeset/fresh-countries-move.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@meilisearch/instant-meilisearch": minor
+---
+
+Replaces search with multiSearch API.
```
Comment (bidoubiwa) [inline .changeset/fresh-countries-move.md:5]: Explain more

**Review by brunoocasali — APPROVED** [review]
LGTM 🎉 multi-search ftw!

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**